whitepaper.tex

```diff
@@ -418,31 +422,71 @@
   systems. \href{https://docs.sel4.systems/projects/available-user-components.html}{We
     have a fair number of these}, but much  more is needed.
 
-  An important enabler is a component framework; it allows developers
-  to focus on the code that implements the services, and automates
-  much of the system integration. There are presently two main
-  component frameworks for seL4, both open source: CAmkES and Genode.
-
-  \href{https://trustworthy.systems/projects/TS/camkes/}{CAmkES} is a
-  framework that is aimed at embedded and cyber-physical systems,
-  which typically have a static architecture, meaning they consist of
-  a defined set of components that does not change once the system has
-  fully booted up.
+  Even compared to other microkernels, seL4's API is very low-level,
+  with only the minimum abstraction as required to securely manage the
+  hardware. As such, building systems on seL4 is particularly
+  difficult. A good way to look at this is to think of seL4 as the
+  ``assembly language of operating systems'': very primitive.
+
+  No-one in their right mind would write a complex system, say a
+  database or web server, in assembly language, we use higher-level
+  programming languages for that. These simplify the task by providing
+  higher-level constructs and abstractions, at the expense of losing
+  some of the power the hardware provides. In almost all cases, that
+  is a winning trade-off.
+
+  Similarly, one should not try to build a complex system directly on
+  seL4, but should use a higher-level framework that provides more
+  appropriate abstractions, at the expense of introducing some policy
+  and taming the power of seL4. Specifically, such a framework should allows developers
+  to focus on the code that implements the services, ignore hardware
+  complexities, and automate
+  much of the system integration.
+
+  There are presently three main
+  component frameworks for seL4, all open source: The Microkit, CAmkES
+  and Genode.
+
+  The \href{https://trustworthy.systems/projects/microkit/}{seL4 Microkit} reduces
+  the complex seL4 API to literally a handful of simple abstractions,
+  designed around modules called \emph{protection domains}. It also
```

Comment:
```suggestion
  designed around components called \emph{protection domains}. It also
```

Just to not overload on the wording, I think it should be components which is what we usually use in documentation.